pkg/authn/keychain.go

```diff
@@ -35,7 +35,19 @@ type Keychain interface {
 
 // defaultKeychain implements Keychain with the semantics of the standard Docker
 // credential keychain.
-type defaultKeychain struct{}
+type defaultKeychain struct {
+	configDir string
+}
+
+// SetDefaultKeychainConfigDir sets the client config directory to specified location.
+func SetDefaultKeychainConfigDir(p string) error {
```

Comment:
It's a bit gross, but you could just call:
```
os.Setenv("DOCKER_CONFIG", p)
```

in the caller's code instead of adding this method here. It's not great, but neither is a mutating package-level vars.

I wouldn't really expect a library to want to set this directly, unless you don't have control over the environment :/